### ISSUE TITLE
Asking the number of images generated from openpano

### ISSUE THREAD

**Issue by BJTU-zjc** [cvg/GeoCalib#26]
Hi!
I am using your code to generate the openpano dataset, but the number of images generated differs significantly from the number described in the paper and README.md. According to images_per_pano: 16 in the configuration file, I expected to generate 2325 * 16 = 37200 images, but only 33374 images were generated, which is about 3826 less than expected.

I have checked the code logic and found the following possible reasons:

In the generate_images_from_pano function, the generated images are skipped if they contain too many black pixels (more than 1%). I think it's the main reason for the reduced number of images.
```python
n_pixels = perspective_image.shape[-2] * perspective_image.shape[-1]
valid = (torch.sum(perspective_image.sum(0) == 0) / n_pixels) < 0.01
  if not valid:
        logger.debug(f"Perspective {perspective_name} has too many black pixels.")
        continue
```
Despite my efforts to train the dataset, the results I obtained were lower than those reported in the paper. This is important for us to reproduce and improve these results. Therefore, I would like some advice to better understand the issue:
- Does the number of datasets mentioned in the paper include the skipped images? 
- What steps can I take to maintain the count as illustrated in your paper?

Thank you for your help! 

>

**Comment by veichta** [comment]
Hi, 
The numbers reported in the paper (Appendix F.1) should correspond to the final number of images after the ones with black pixels are removed (37015 train images from 2262 training panoramas).

Did you change anything on the default config? Having generally bigger values for pitch or FoV could lead to more images being removed. 

Another problem could be that we previously pointed to the wrong dataset (#16) which could also be the reason that you have fewer images.

**Comment by BJTU-zjc** [comment]
> Another problem could be that we previously pointed to the wrong dataset ([#16](https://github.com/cvg/GeoCalib/issues/16)) which could also be the reason that you have fewer images.

Yes， we already addressed this problem and generated the train_pano.txt which includes 2325 images. 

> Did you change anything on the default config? Having generally bigger values for pitch or FoV could lead to more images being removed.

Thank you for your suggestion. I checked the config again, and it is the same as you published:
```python
name: openpano
base_dir: data/openpano
pano_dir: "${.base_dir}/panoramas"
images_per_pano: 16
resize_factor: null
n_workers: 1
device: cuda
overwrite: true
parameter_dists:
  roll:
    type: uniform # uni[-45, 45]
    options:
      loc: -0.7853981633974483 # -45 degrees
      scale: 1.5707963267948966 # 90 degrees
  pitch:
    type: uniform # uni[-45, 45]
    options:
      loc: -0.7853981633974483 # -45 degrees
      scale: 1.5707963267948966 # 90 degrees
  vfov:
    type: uniform # uni[20, 105]
    options:
      loc: 0.3490658503988659 # 20 degrees
      scale: 1.48352986419518 # 85 degrees
  resize_factor:
    type: uniform
    options:
      loc: 1.2
      scale: 0.5
  shape:
    type: fix
    value:
      - 640
      - 640
```
In addition, I modified the log output in the function `generate_images_from_pano` so that when an image contains too many black pixels it is logged:

```python 
def __init__(self, conf):
      self.skipped_count = 0
      ...
def generate_images_from_pano: 
      ...
      for i, perspective_image in enumerate(perspective_images):
            perspective_name = f"{pano_path.stem}_{i}.jpg"

            n_pixels = perspective_image.shape[-2] * perspective_image.shape[-1]
            valid = (torch.sum(perspective_image.sum(0) == 0) / n_pixels) < 0.01
            if not valid:
                logger.debug(f"Perspective {perspective_name} has too many black pixels.")
                self.skipped_count += 1
                continue

            write_image(perspective_image, out_dir / perspective_name)
      ...
def generate_split:
      ...
      if self.skipped_count > 0:
            logger.info(f"Skipped {self.skipped_count} images due to too many black pixels.")
```
and the output is:
```python
[2025-03-08 06:00:40,665][__main__][INFO] - Skipped 4237 images due to too many black pixels.
[2025-03-08 06:00:40,678][__main__][INFO] - Generated 33374 train images.
[2025-03-08 06:00:40,679][__main__][INFO] - Generated 2193 val images.
[2025-03-08 06:00:40,679][__main__][INFO] - Generated 2212 test images.
```
Thanks again for your reply and let me know if you have any other suggestions!

**Comment by BJTU-zjc** [comment]
Hi, @veichta 

Thank you for your suggestions and the discussion in issue #16. I realized that I had mistakenly downloaded the Laval Indoor HDR Dataset instead of the Laval Indoor Photometric HDR Dataset.

However, after regenerating the training dataset using train_pano.txt, I found that the number of images is still smaller than the GeoCalib training dataset, and the filenames in pano_train.txt do not match the expected count of 2612. The validation and test set sizes do match the paper.

Would it be possible for you to ensure the pano_train.txt is correct? Or could you tell me what the number of images in train_pano.txt represents and why it differs from the training dataset size mentioned in the paper? Otherwise, I may have to supplement it with additional panorama images to match the paper’s requirements as closely as possible.

I appreciate your help!

**Comment by veichta** [comment]
Hi, how many panoramas do you have in total? Do they now roughly match the number from the paper (2912)?

**Comment by BJTU-zjc** [comment]
Now, I have the Laval Indoor Photometric HDR Dataset containing 2362 images. The openpano/panoramas/train folder contains 316 images, the openpano/panoramas/val folder has 39 images, and the openpano/panoramas/test folder includes 40 images. The total count of images is 2757.

**Comment by CharalamposTzamos** [comment]
I am getting the same number of images as @BJTU-zjc for training. Would this cause an issue in the training? I have verified that the number of downloaded paranoramas from Laval Indoor Photometric HDR Dataset is correct, for the rest I used the provided scripts for tonemapping, downloading HDRMAPS and PolyHaven, and preparing the cropped images for training, validation, and testing.

**Comment by tonghaiping** [comment]
> 现在，我拥有包含 2362 幅图像的 Laval 室内光度 HDR 数据集。openpano/panoramas/train 文件夹包含 316 幅图像，openpano/panoramas/val 文件夹包含 39 幅图像，openpano/panoramas/test 文件夹包含 40 幅图像。图像总数为 2757 幅。
Laval Indoor Photometric HDR Dataset how to get this dataset

**Comment by veichta** [comment]
Hi, we have noticed that there are indeed some panoramas missing in the zip that we share. While this doesn't seem to significantly impact the overall result, please grab the corrected version from this link for now: https://polybox.ethz.ch/index.php/s/XK4oM1l6ZqSIXw9

I have updated the readme and download scrips accordingly. 

The main website link will be updated as soon as possible.

**Comment by tonghaiping** [comment]
After running the script to process the data, my data volume does not match the author's paper. The result of my data generation is:
[2025-07-01 11:48:49,307][__main__][INFO] - Config:
name: openpanov2
base_dir: data/openpanov2
pano_dir: ${.base_dir}/panoramas
pano_train: ${.pano_dir}/train
pano_val: ${.pano_dir}/val
pano_test: ${.pano_dir}/test
perspective_dir: ${.base_dir}/${.name}
perspective_train: ${.perspective_dir}/train
perspective_val: ${.perspective_dir}/val
perspective_test: ${.perspective_dir}/test
train_csv: ${.perspective_dir}/train.csv
val_csv: ${.perspective_dir}/val.csv
test_csv: ${.perspective_dir}/test.csv
camera_model: pinhole
parameter_dists:
  roll:
    type: uniform
    options:
      loc: -0.7853981633974483
      scale: 1.5707963267948966
  pitch:
    type: uniform
    options:
      loc: -0.7853981633974483
      scale: 1.5707963267948966
  vfov:
    type: uniform
    options:
      loc: 0.3490658503988659
      scale: 1.48352986419518
  resize_factor:
    type: uniform
    options:
      loc: 1.2
      scale: 0.5
  shape:
    type: fix
    value:
    - 640
    - 640
images_per_pano: 16
n_workers: 1
device: cuda
overwrite: true
resize_factor: null

[2025-07-01 11:48:49,332][__main__][INFO] - Writing perspective images to data/openpanov2/openpanov2/train
[2025-07-01 14:08:15,604][__main__][INFO] - Writing perspective images to data/openpanov2/openpanov2/val
[2025-07-01 14:16:20,000][__main__][INFO] - Writing perspective images to data/openpanov2/openpanov2/test
[2025-07-01 14:24:30,524][__main__][INFO] - Generated 38139 train images.
[2025-07-01 14:24:30,524][__main__][INFO] - Generated 2148 val images.
[2025-07-01 14:24:30,525][__main__][INFO] - Generated 2174 test images.
